app/forms/mappings/Constraints.scala

```diff
@@ -156,4 +158,13 @@ trait Constraints {
       case _ =>
         Invalid(errorMsg)
     }
+
+  def countryAlreadySelected(errorMsg: String, userAnswers: UserAnswers): Constraint[String] =
+    Constraint {
+      case countryCode
+        if !userAnswers.get(IncomeSourceCountries).toSeq.flatten.contains(Country(CountryNamesDataSource.getCountryName(countryCode), countryCode)) =>
```

Comment:
The if condition could be
```
if !userAnswers.get(IncomeSourceCountries).toSeq.flatten.map(_.code).contains(countryCode)
```
to avoid having to retrieve and compare the country name